pkg/rasactl/connect_rasa.go

```diff
@@ -0,0 +1,302 @@
+	fileEndpoints := fmt.Sprintf("%s/.endpoints.yaml", configDir)
+
+	mutualArgs := []string{
+		"run",
+		"--verbose",
+		"--enable-api",
+		"--cors",
+		"'*'",
+		"--auth-token",
+		rasaToken,
+		"--credentials",
+		fileCreds,
+		"--endpoints",
+		fileEndpoints,
+	}
+	var productionPort int = r.Flags.ConnectRasa.Port
+	var workerPort int = r.Flags.ConnectRasa.Port
+
+	if r.Flags.ConnectRasa.RunSeparateWorker {
+		workerPort = workerPort + 1
+		environmentName = "production"
+	}
+
+	if len(r.Flags.ConnectRasa.ExtraArgs) != 0 {
+		mutualArgs = append(mutualArgs, r.Flags.ConnectRasa.ExtraArgs...)
+	}
+
+	r.Log.Info("Connecting Rasa Server to Rasa X")
+	if r.KubernetesClient.GetBackendType() == types.KubernetesBackendLocal {
+		r.HelmClient.SetValues(utils.MergeMaps(helm.ValuesRabbitMQNodePort(),
+			helm.ValuesPostgreSQLNodePort(), helm.ValuesHostNetworkRasaX(), r.HelmClient.GetValues()))
+		r.Log.V(1).Info("Merging values", "result", r.HelmClient.GetValues())
+	} else {
+		return errors.Errorf(
+			"It looks like you're not using kind as a backend for Kubernetes cluster, the connect rasa command is available only if you use kind.",
+		)
+	}
+
+	r.Log.Info("Upgrading configuration for Rasa X deployment")
+	if err := r.HelmClient.Upgrade(); err != nil {
+		return err
+	}
+
+	r.Log.Info("Updating configuration for Rasa X")
+	if err := r.KubernetesClient.UpdateRasaXConfig(rasaToken); err != nil {
+		return err
+	}
+
+	r.Log.Info("Restarting Rasa X pod")
+	if err := r.KubernetesClient.DeleteRasaXPods(); err != nil {
+		return err
+	}
+
+	if err := r.saveRasaCredentialsFile(fileCreds); err != nil {
+		return err
+	}
+
+	if err := r.saveRasaEndpointsFile(fileEndpoints); err != nil {
+		return err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*360)
+	defer cancel()
+	url, err := r.GetRasaXURL()
+	if err != nil {
+		return err
+	}
+	r.initRasaXClient()
+	r.RasaXClient.URL = url
+	if err := r.RasaXClient.WaitForDatabaseMigration(ctx); err != nil {
+		return err
+	}
+
+	msg := "Starting Rasa Server"
+	r.Spinner.Message(msg)
+	r.Log.Info(msg, "args", mutualArgs)
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGINT, syscall.SIGTERM)
+	ready := make(chan bool, 1)
+	go func() {
+		sig := <-sigs
+		fmt.Println()
+		fmt.Println(sig)
+		ready <- true
+	}()
+
+	go func(ready chan bool) {
+		done := make(chan bool, 1)
+		args := mutualArgs
+		args = append(args,
+			"-p",
+			fmt.Sprintf("%d", productionPort),
+		)
+		r.runRasaServer(environmentName, args, done, ready)
+		<-done
+	}(ready)
+
+	if r.Flags.ConnectRasa.RunSeparateWorker {
+		r.Log.Info("Running separate Rasa X server for the worker environment")
+		go func(ready chan bool) {
+			done := make(chan bool, 1)
+			args := mutualArgs
+			args = append(args,
+				"-p",
+				fmt.Sprintf("%d", workerPort),
+			)
+			r.runRasaServer("worker", args, done, ready)
+			<-done
+		}(ready)
+	}
+	r.Spinner.Stop()
+	<-ready
```

Comment:
Should the `Stop` go after the channel receive instead? Otherwise we might stop the spinner and then wait some time after receiving from `ready`.

Reply:
All looks good, `ready` is send when a `SIGINT` is received. Maybe the name of the channel is not the best, but it's waiting for `SIGINT` or `SIGTERM`.